ODIN_II/SRC/netlist_create_from_ast.cpp

```diff
@@ -2755,10 +2755,24 @@ signal_list_t* create_output_pin(ast_node_t* var_declare, char* instance_name_pr
 /*---------------------------------------------------------------------------------------------
  * (function: assignment_alias)
  *-------------------------------------------------------------------------------------------*/
-signal_list_t* assignment_alias(ast_node_t* assignment, char* instance_name_prefix, sc_hierarchy* local_ref, long assignment_size) {
+signal_list_t* assignment_alias(ast_node_t* assignment, char* instance_name_prefix, sc_hierarchy* local_ref, long assignment_size, ids parent_ids) {
     ast_node_t* left = assignment->children[0];
     ast_node_t* right = assignment->children[1];
 
+    /* For the check - Wire cannot be on left in always block */
+    if (parent_ids != ASSIGN) {
+        /* TODO: Individual objects to be checked for wire in case of CONCATENATE */
+        if ((left->type != RANGE_REF) && (left->type != ARRAY_REF) && (left->type != CONCATENATE)) {
+            STRING_CACHE* local_symbol_table_sc = local_ref->local_symbol_table_sc;
```

Comment:
This is a good approach @harpreetbamrah , thanks!
However, the structure of the code is not like using the information from the parents since it is not actually needed. That means the hierarchy is from up to down, in no place of the code you would find a similar thing. 
So, I would recommend you to specify the type of the wire in the beginning of the code instead of resolving it later in netlist creation phase. In this way there is no need to deal with this much information.

Reply:
When we read the module port, we specify their type at the same time. So you can simply consider the default type as wire when there is no type declaration.
please look at the default case of the first switch case in function `markAndProcessPortWith` at parse_making_ast.
there is actually where we specify the type of each net.
https://github.com/CAS-Atlantic/vtr-verilog-to-routing/blob/e3be0b29ab08b951f6962b3231481f200636427f/ODIN_II/SRC/parse_making_ast.cpp#L468-L499